addons/mail/static/src/discuss/core/common/discuss_core_common_service.js

```diff
@@ -198,7 +198,7 @@ export class DiscussCoreCommon {
         message = this.store.Message.insert(messageData, { html: true });
         if (message.notIn(channel.messages)) {
             if (!channel.loadNewer) {
-                channel.messages.push(message);
+                channel.replaceMessages(this.store.Message.get(temporaryId), message);
```

Comment:
What bug is this change handling actually?

Reply:
Shouldn't these fixes be done in earlier versions too?

Reply:
To not let the message from odooBot (_message_post_after_hook) be inserted before the genuine message.